Library/Homebrew/formula_installer.rb

```diff
@@ -711,7 +718,7 @@ def build
   end
 
   def link(keg)
-    if formula.keg_only?
+    unless @keg_was_linked
```

Comment:
@reitermarkus This logic is badly broken but in general: no because we're trying to maintain link status, keg-only or not.